mappings/clacso.py

```diff
@@ -1,23 +1,119 @@
+            'title': ('./dc:title/text()', '{0}'),
+            'authors': [('./dc:creator/text()', '{0}|||true')],
+            'abstract': ('./dc:description/text()', '{0}'),
+            'dates': [
+                (
+                    [
+                        './dc:date/text()',
+                    ],
+                    '{0}'
+                )
+            ],
+            'identifiers': [
+                ('./dc:identifier/text()', '{0}'),
+            ],
+            'medium' : [
+                ('./dc:type/text()', '{0}')
+            ],
+            'has_part': [(
+                './dc:identifier/text()',
+                '1|{0}|clacso|text/html|{{"reader": false, "download": false, "catalog": false, "embed": true}}'
+            )]
         }
 
     def applyFormatting(self):
         self.record.source = 'clacso'
-        self.record.source_id = f'{self.record.source}'
+        handle_url = "https://biblioteca-repositorio.clacso.edu.ar/handle/CLACSO"
+        self.formatting_source_id(handle_url)
+        self.formatting_medium()
+        self.formatting_dates()
+        self.formatting_identifiers()
+        self.formatting_has_part(handle_url)
+
+    def formatting_source_id(self, handle_url):
+        for identifier in self.record.identifiers:
+            if handle_url in identifier:
+                id = identifier.split('/')[-1]
+                self.record.source_id = f'{self.record.source}|{id}'
+                break
+
+    def formatting_medium(self):
+        if self.record.medium:
+            type_list = ['book', 'bookpart', 'part', 'chapter', 'bibliography', 'appendix', 'index',
+                    'foreword', 'afterword', 'bibliography', 'review', 'article', 'introduction']
+            for medium in self.record.medium:
+                if '/' in medium:
+                    medium_value = medium.split('/')[-1]
+                    if medium_value.lower() in type_list:
+                        self.record.medium = medium_value.lower()
+                        break
+                elif '/' not in medium and medium.lower() in type_list:
+                    medium_value = medium
+                    self.record.medium = medium_value.lower()
+                    break
+        else:
+            self.record.medium = ''
+
+    def formatting_dates(self):
+        if self.record.dates:
+            year_list = []
+            for date in self.record.dates:
+                if '-' in date:
+                    year_list.append(date.split('-')[0])
+                else:
+                    year_list.append(date)
+        self.record.dates = [f'{min(year_list)}|issued']
+
+    def formatting_identifiers(self):
+        if self.record.identifiers:
+            for index, identifier in enumerate(self.record.identifiers):
+                count_digits = 0
+                if '/' not in identifier:
+                    for char in identifier:
```

Comment:
Can we pull this logic for determining if an identifier is ISSN or ISBN into utils functions to be reused in other mappings?

Reply:
Sure. I can put this method into the `base_mapping` just like `applyFormatting` for all the mappings.